docs/usage/configuration-options.md

```diff
@@ -173,6 +173,25 @@ Example use:
 }
 ```
 
+## automergeStrategy
+
+This setting is only applicable if you opt-in by configuring `automerge` to `true` and `automergeType` to `pr` for any of your dependencies.
+
+The automerge strategy defaults to `auto`, in which Renovate will make its best guess as to how to merge pull requests.
+This generally results in Renovate respecting the strategy configured in the platform itself for the repository if possible.
+Acceptable values are:
+
+- `auto`, in which the choice is left to Renovate
+- `fast-forward`, which generally involves no new commits in the resultant tree, but "fast-forwarding" the main branch reference
+- `merge-commit`, which generally involves synthesizing a new merge commit
+- `rebase`, which generally involves rewriting history as part of the merge — but usually retaining the individual commits
```

Comment:
The way Renovate rebases **is not the same** as the way Git rebases.

Git rebases by making new commits and putting those on top of the rebased branch.

Renovate "rebases" by just doing a new run and force pushes a single new commit.
So it does not "retain individual commits". 😉 

```suggestion
- `rebase`, Renovate starts a fresh run and pushes a single new commit, that's up-to-date with the target branch
```

Maybe something like this is better? Not 100% sure of the correct wording. 😄

Reply:
Hm yes, I see. This `rebase` is outside of renovate's usually rebase behaviour, and I added it here because most of the other platforms support something that looks like a merge rebase. I see two ways forward here:

1) Remove it. The only platform where this is exposed doesn't support a rebase-alike anyway.
2) Change it to `merge-rebase`, so as not to conflict with the usage of "rebase" that Renovate does when it does git directly.

I added it as a general thing because other platforms support it to various levels, but it might eventually become more complicated anyway — some platforms like Bitbucket Server support both `rebase + merge --no-ff` as well as `rebase + merge --ff-only` as explicit choices, and "rebase" alone isn't enough to describe this. The description I've added here attempted to be vague about these differences, committing only to "And there might be some history rewriting...". However, in these merge strategies, the commit contents do tend to keep existing, even if the history is rewritten.

Reply:
This is going over my head, I know and use `git rebase -i upstream/main` to fixup/squash/drop commits, on my feature branches to clean up my work before I push it. But I'm not familiar with all the ways in which different platforms handle rebasing.

I'll defer to @rarkins's judgement on the wording here, as I'm not qualified to choose here. 😉

Reply:
Oh by the way, we do have an issue (https://github.com/renovatebot/renovate/issues/9020) to replace the term `rebase` with `refresh`. Maybe the word `refresh` could be useful here as well?

Reply:
Again, it's not our (IE, Renovate's) `rebase` terminology here, it's the `rebase` as referred to by the platforms that are hosting the repository. For example:

* Bitbucket Server has "rebase" strategies outlined in https://confluence.atlassian.com/bitbucketserver/pull-request-merge-strategies-844499235.html#Pullrequestmergestrategies-Mergestrategies.
* Github has a "rebase" strategy outlined in https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#rebase-and-merge-your-pull-request-commits
* Gitea has a "rebase" strategy added in https://github.com/go-gitea/gitea/pull/3188

None of these platforms refer to these merge strategies as "refresh strategies". The use of "refresh" inside Renovate to refer to the process by which Renovate keeps branches up to date is nice, but completely orthogonal to how "rebase" is used here.

Reply:
@rarkins, Could I get you to have a read through of this comment chain specifically and answer @HonkingGoose and mine's discussion?

Reply:
@HonkingGoose We should leave as is, as this `renbase` means which option the platform uses to merge the pr. It has nothing to do with the renovate way of recreate fresh commit and force-push.

@jbirch-atlassian I think we can leave as is.

Reply:
Oh, if we're talking about the platforms `rebase` strategy, then we can keep things as-is. 😉